app/src/androidTest/java/org/mozilla/fenix/ui/robots/SearchRobot.kt

```diff
@@ -178,4 +215,54 @@ fun searchScreen(interact: SearchRobot.() -> Unit): SearchRobot.Transition {
     return SearchRobot.Transition()
 }
 
+private fun assertKeyboardVisibility(isExpectedToBeVisible: Boolean) = {
+    mDevice.waitNotNull(
+        Until.findObject(
+            By.text("Search Engine")
+        ), waitingTime
+    )
+    assertEquals(
+        isExpectedToBeVisible,
+        UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+            .executeShellCommand("dumpsys input_method | grep mInputShown")
+            .contains("mInputShown=true")
+    )
+}
+
+private fun assertSearchEngineList() {
+    scrollToElementByText("Google")
+    onView(withText("Google"))
```

Comment:
Remove all these scrollToElementByText. Instead on line 233 add `onView(withId(R.id.mozac_browser_toolbar_edit_icon)).click()
`
Thay way the keyboard will be dismissed and we can check all the elements on the screen without having to scroll.